### ISSUE TITLE
manual sysext merge control

### ISSUE THREAD

**Issue by hsitter** [systemd/systemd#25031]
### Component

systemd-sysext

### Is your feature request related to a problem? Please describe

At KDE we are currently looking to drive our dev session builds through systemd-sysext (https://invent.kde.org/plasma/plasma-workspace/-/merge_requests/1715). One of the things we'd really like to do is merge our extension only when necessary and unmerge when no longer necessary.

To give some background: dev session here refers to an entire software stack installed in the developer's home directory. The sysext of this installed tree is meant to "overlay" the distribution provided KDE software when the developer logs into the dev session via their display manager (it has a distinct session file). When they log out of the session they should still be able to log into the regular distro provided session again.

### Describe the solution you'd like

Giving more merge/unmerge control would do nicely. To that end the extension-release file would probably grow a boolean option to not merge the extension automatically `AUTO_MERGE=false` such that one needs to explicitly call `systemd-sysext merge foobar` and `systemd-sysext unmerge foobar` to control the state of the extension.

### Describe alternatives you've considered

Another option would be to have sysextensions actually tied to logind sessions such that systemd itself takes care of merge on login and unmerge on logout. Probably more work for not that much gain though.

### The systemd version you checked that didn't have the feature you are asking for

251

**Comment by poettering** [comment]
makes sense I guess. I guess a new parameter to override the list of images to apply (and disable auto-discovery) would make a lot of sense, and would be pretty trivial to add.

**Comment by poettering** [comment]
so, here's another idea though, which might work better in the long run:

keep your images outside of the usual image search path, and only symlink it in there when you want it to be active. i.e. link the image into `/run/extensions/` or `/etc/extensions/` when you want to activate it, and unlink it from there when you don't.

Benefit of this scheme: you won't affect other potential users of sysext, every tool like yours can safely add/remove sysexts for activation that way.

I am pretty sure such a scheme should just work already, but if not I'd consider that a bug we should fix.

If you want to enable your sysext image across boots, then you'd use `/etc/extensions/`, but if you only want to do so temporarily without persistence, then `/run/extensions/` is the place to go.
